lib/src/main/java/org/exoplatform/social/addons/updater/IdentityMigrationService.java

```diff
@@ -276,19 +277,25 @@ public void doRemove() throws Exception {
     } finally {
       MigrationContext.setIdentitiesCleanupFailed(identitiesCleanupFailed);
       if (identitiesCleanupFailed.size() > 0) {
-        LOG.warn("Cleanup failed for " + identitiesCleanupFailed.size() + " identities");
+        LOG.warn("Cleanup failed for {} identities", identitiesCleanupFailed.size());
       }
-      LOG.info(String.format("| / END::cleanup Identities migration for (%s) identity consumed %s(ms)", offset, System.currentTimeMillis() - t));
+      LOG.info("| / END::cleanup ({}) identities duration_ms={}", offset, System.currentTimeMillis() - t);
     }
   }
 
   private Identity migrateIdentity(Node node, String jcrId) throws Exception {
     String providerId = node.getProperty("soc:providerId").getString();
     String remoteId = node.getProperty("soc:remoteId").getString();
+    String name = node.getName();
+
+    // Ensure the nodes are coherents
+    if (!name.equals("soc:" + remoteId)) {
+      throw new Exception(String.format("Identity has different name and remote_id provider_id=%s name=%s remote_id=%s", providerId, remoteId, name));
```

Comment:
Yes it's the goal in the case we found, 2 spaces content were inverted after the migration due to this incoherency but it can also result to an space without identity. 
We agreed with @thomasdelhomenie, as it's a case that should not happen in "normal" conditions, to block the migration to avoid data corruption.

An other solution would be to review the data model to add an "hard link" between the space itself and the space identity. Even if such a link need probably be added in the model, it looks overkill to do it for this case.

@thomasdelhomenie WDYT ?